hibernate-reactive-core/src/test/java/org/hibernate/reactive/SchemaUpdateSqlServerTestBase.java

```diff
@@ -0,0 +1,403 @@
+													context.assertEquals( false, ( (Object[]) list.get(0) )[1] );
+												} )
+										)
+										.thenCompose( v -> s.createNativeQuery( foreignKeyDefinitionQuery )
+												.setParameter( 1, "fk_asimple_aother" )
+												.setParameter( 2, "ASimple" )
+												.setParameter( 3, "AOther" )
+												.getResultList()
+												.thenAccept( results -> {
+													context.assertEquals( 2, results.size() );
+													context.assertEquals( "id1", ( (Object[]) results.get( 0 ) )[0] );
+													context.assertEquals( "id1", ( (Object[]) results.get( 0 ) )[1] );
+													context.assertEquals( "id2", ( (Object[]) results.get( 1 ) )[0] );
+													context.assertEquals( "id2", ( (Object[]) results.get( 1 ) )[1] );
+												} )
+										)
+										.thenCompose( v -> s.createNativeQuery( foreignKeyDefinitionQuery )
+												.setParameter( 1, "fk_asimple_aanother" )
+												.setParameter( 2, "ASimple" )
+												.setParameter( 3, "AAnother" )
+												.getSingleResult()
+												.thenAccept( result -> {
+													context.assertEquals( "aAnother_id", ( (Object[]) result )[0] );
+													context.assertEquals( "id", ( (Object[]) result )[1] );
+												} )
+										)
+								) )
+		);
+
+	}
+
+	@Entity(name = "ASimple")
+	@Table(name = "ASimple", indexes = @Index(
+			name = "i_asimple_avalue_astringValue",
+			columnList = "aValue ASC, aStringValue DESC"
+	))
+	public static class ASimpleFirst {
+		@Id
+		@GeneratedValue
+		private Integer id;
+		private Integer aValue;
+		private String aStringValue;
+		@ManyToOne(cascade = CascadeType.ALL)
+		@JoinColumns(
+				value = { @JoinColumn(name = "id1"), @JoinColumn(name = "id2") },
+				foreignKey = @ForeignKey(name = "fk_asimple_aother")
+		)
+		private AOther aOther;
+	}
+
+	@Entity(name = "ASimple")
+	@Table(name = "ASimple", indexes = {
+			@Index(name = "i_asimple_avalue_astringValue", columnList = "aValue ASC, aStringValue DESC"),
+			@Index(name = "i_asimple_avalue_data", columnList = "aValue DESC, data ASC")
+	},
+			uniqueConstraints = { @UniqueConstraint(name = "u_asimple_astringvalue", columnNames = "aStringValue") }
+	)
+	public static class ASimpleNext {
+		@Id
+		@GeneratedValue
+		private Integer id;
+
+		private Integer aValue;
+
+		private String aStringValue;
+
+		private String data;
+
+		@ManyToOne(cascade = CascadeType.ALL)
+		@JoinColumns(
+				value = { @JoinColumn(name = "id1"), @JoinColumn(name = "id2") },
+				foreignKey = @ForeignKey(name = "fk_asimple_aother")
+		)
+		private AOther aOther;
+
+		@ManyToOne(cascade = CascadeType.ALL)
+		@JoinColumn(foreignKey = @ForeignKey(name = "fk_asimple_aanother"))
+		private AAnother aAnother;
+	}
+
+	@Entity(name = "AOther")
+	@IdClass(AOtherId.class)
+	public static class AOther {
+		@Id
+		private int id1;
+
+		@Id
+		private String id2;
+
+		private String anotherString;
+	}
+
+	public static class AOtherId implements Serializable {
```

Comment:
@gbadner I've just noticed this class but I don't think we are using it. Is it necessary for the test or is this a leftover?

Reply:
@DavideD, yes, it is needed. It is an `IdClass` used by `AOther`.

Reply:
Ah yes, sorry